helix-term/src/commands.rs

```diff
@@ -3943,96 +4022,111 @@ fn goto_prev_comment(cx: &mut Context) {
-                            let ch = text.char(range.cursor(text));
-                            if !ch.is_ascii_alphanumeric() {
-                                textobject::textobject_surround(text, range, objtype, ch, count)
-                            } else {
-                                range
-                            }
-                        }
-                        // TODO: cancel new ranges if inconsistent surround matches across lines
-                        ch if !ch.is_ascii_alphanumeric() => {
-                            textobject::textobject_surround(text, range, objtype, ch, count)
-                        }
-                        _ => range,
-                    }
-                });
-                doc.set_selection(view.id, selection);
-            };
-            textobject(cx.editor);
-            cx.editor.last_motion = Some(Motion(Box::new(textobject)));
-        }
-    });
+select_textobject_commands! {
+    select_around_word(Around, Word(false));
+    select_inside_word(Inside, Word(false));
+    select_around_long_word(Around, Word(true));
+    select_inside_long_word(Inside, Word(true));
+    select_around_class(Around, Treesitter("class"));
+    select_inside_class(Inside, Treesitter("class"));
+    select_around_function(Around, Treesitter("function"));
+    select_inside_function(Inside, Treesitter("function"));
+    select_around_parameter(Around, Treesitter("parameter"));
+    select_inside_parameter(Inside, Treesitter("parameter"));
+    select_around_cursor_pair(Around, Matching(None));
+    select_inside_cursor_pair(Inside, Matching(None));
+}
 
-    if let Some((title, abbrev)) = match objtype {
-        textobject::TextObject::Inside => Some(("Match inside", "mi")),
-        textobject::TextObject::Around => Some(("Match around", "ma")),
-        _ => return,
-    } {
-        let help_text = [
-            ("w", "Word"),
-            ("W", "WORD"),
-            ("c", "Class (tree-sitter)"),
-            ("f", "Function (tree-sitter)"),
-            ("a", "Argument/parameter (tree-sitter)"),
-            ("o", "Comment (tree-sitter)"),
-            ("m", "Matching delimiter under cursor"),
-            (" ", "... or any character acting as a pair"),
-        ];
+fn prompt_and_select_around_pair(cx: &mut Context) {
+    cx.editor.set_status("Select a delimiter...");
+    cx.on_next_key(select_around_pair);
+}
 
-        cx.editor.autoinfo = Some(Info::new(
-            title,
-            help_text
-                .into_iter()
-                .map(|(col1, col2)| (col1.to_string(), col2.to_string()))
-                .collect(),
-        ));
-        cx.editor.pseudo_pending = Some(abbrev.to_string());
-    };
+fn prompt_and_select_inside_pair(cx: &mut Context) {
+    cx.editor.set_status("Select a delimiter...");
+    cx.on_next_key(select_inside_pair);
+}
+
+fn select_around_pair(cx: &mut Context, event: KeyEvent) {
+    select_pair(cx, textobject::TextObject::Around, event);
+}
```

Comment:
I'd rather this be called `select_around_pair_fallback` and `prompt_and_select_around_pair` be called `select_around_pair`.